Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/eventregistration/attendee/checkins/CheckInsFragment.kt

```diff
@@ -57,6 +58,12 @@ class CheckInsFragment : Fragment(R.layout.trace_location_attendee_checkins_frag
             adapter = checkInsAdapter
             addItemDecoration(TopBottomPaddingDecorator(topPadding = R.dimen.spacing_tiny))
             itemAnimator = DefaultItemAnimator()
+            onSwipeItem(
+                context = requireContext(),
+                excludedPositions = listOf() // TODO exclude items from swiping such as Camera permission item
+            ) { position, _ ->
+                showRemovalConfirmation(checkInsAdapter.data[position].checkin)
+            }
```

Comment:
What do you think about leaving this up to each item?
e.g.

```kotlin
(data[position] as? SwipeConsumer)?.let { it.onSwipe(SwipeConsumer.SWIPE_LEFT) }
// ...
data class Item(....) : SwipeConsumer
```
Then each item that wants to do something on swipe just has to implement the interface, and can directly call a callback that we set in the viewmodel.

Reply:
Good Idea!